python/mpt/src/mpt/hash_diff.cairo

```diff
@@ -65,11 +65,25 @@ func poseidon_storage_diff{poseidon_ptr: PoseidonBuiltin*}(diff: StorageDiffEntr
 
     assert buffer[0] = diff.value.key.value;
     // Inline the prev storage value (U256)
-    assert buffer[1] = diff.value.prev_value.value.low;
-    assert buffer[2] = diff.value.prev_value.value.high;
+    if (cast(diff.value.prev_value.value, felt) == 0) {
+        // Note: in case of State Diffs, writing the default U256(0) value deletes the key, writing a null pointer instead.
+        // In that case, we hash (0, 0) as the value.
+        assert buffer[1] = 0;
+        assert buffer[2] = 0;
+    } else {
+        assert buffer[1] = diff.value.prev_value.value.low;
+        assert buffer[2] = diff.value.prev_value.value.high;
+    }
```

Comment:
```suggestion
    assert buffer[1] = diff.value.prev_value.value.low;
    assert buffer[2] = diff.value.prev_value.value.high;
```

this shouldn't ever happen because the dict is a default_dict with de
fault_value U256(0), so if the value is not present before, the `prev` is `U256(0)`